jenkins/job-configs/kubernetes-jenkins/kubernetes-upgrade.yaml

```diff
@@ -226,6 +226,101 @@
+        - 'kubernetes-e2e-{suffix}':
+            suffix: 'gke-{image-old}-{version-old}-{image-new}-{version-new}-upgrade-cluster-new'
+            description: 'Deploys a cluster at v{version-old} at {image-old}, upgrades the master to v{version-new} at {image-new}, and runs the v{version-old} tests.'
+            timeout: 600
+            job-env: |
+                export E2E_OPT="--check_version_skew=false"
+                export E2E_UPGRADE_TEST="true"
+                export GINKGO_UPGRADE_TEST_ARGS="--ginkgo.focus=\[Feature:ClusterUpgrade\] --upgrade-target=ci/{version-new} --upgrade-image={image-new}"
+                export JENKINS_PUBLISHED_SKEW_VERSION="ci/{version-new}"
+                export JENKINS_PUBLISHED_VERSION="ci/{version-old}"
+                export JENKINS_USE_SKEW_TESTS="true"
+                export KUBE_NODE_OS_DISTRIBUTION="{image-old}"
+                export PROJECT="gke-up-{version-infix}-up-clu-n"
+                {version-env}
+
+# This is the job configuration for GCE upgrade tests for GCI and between container-vm and GCI.
+- project:
+    name: kubernetes-e2e-upgrades-gke-cross-image-and-gci
+    jobs:
+        - 'kubernetes-e2e-gke-{image-old}-{version-old}-{image-new}-{version-new}':  # kubernetes-e2e-gke-debian-latest-1.3-gci-latest
+            version-old: 'latest-1.3'
+            version-new: 'latest'
+            version-infix: 'c1-3-glat'
+            image-old: 'debian'
+            image-new: 'gci'
+            version-env: ''
+        - 'kubernetes-e2e-gke-{image-old}-{version-old}-{image-new}-{version-new}':  # kubernetes-e2e-gke-gci-latest-1.3-debian-latest
+            version-old: 'latest-1.3'
+            version-new: 'latest'
+            version-infix: 'g1-3-clat'
+            image-old: 'gci'
+            image-new: 'debian'
+            version-env: ''
+        - 'kubernetes-e2e-gke-{image-old}-{version-old}-{image-new}-{version-new}':  # kubernetes-e2e-gke-gci-latest-1.3-gci-latest
+            version-old: 'latest-1.3'
+            version-new: 'latest'
+            version-infix: 'g1-3-glat'
+            image-old: 'gci'
+            image-new: 'gci'
+            version-env: ''
+        - 'kubernetes-e2e-gke-{image-old}-{version-old}-{image-new}-{version-new}':  # kubernetes-e2e-gke-debian-latest-1.4-gci-latest
+            version-old: 'latest-1.4'
+            version-new: 'latest'
+            version-infix: 'c1-4-glat'
+            image-old: 'debian'
+            image-new: 'gci'
+            version-env: ''
+        - 'kubernetes-e2e-gke-{image-old}-{version-old}-{image-new}-{version-new}':  # kubernetes-e2e-gke-gci-latest-1.4-debian-latest
+            version-old: 'latest-1.4'
+            version-new: 'latest'
+            version-infix: 'g1-4-clat'
+            image-old: 'gci'
+            image-new: 'debian'
+            version-env: ''
+        - 'kubernetes-e2e-gke-{image-old}-{version-old}-{image-new}-{version-new}':  # kubernetes-e2e-gke-gci-latest-1.4-gci-latest
```

Comment:
You also need a configuration for
`
image-old: 'container_vm'
image-new: 'container_vm'
`